### PR TITLE
fix: Spark HBase connector on aarch64

### PR DESCRIPTION
**Pull request by siegfriedweber**
# Description

Fix the build of the Spark HBase connector on aarch64.

For aarch64, the protoc artifact from com.github.os72 is used. The same was already done for Hive (https://github.com/stackabletech/docker-images/blob/24.7.0/hive/stackable/patches/3.1.3/002-HIVE-21939-3.1.3.patch).

The "Apache Hadoop YARN Application Catalog Webapp" required an update of the node version to 14.17.0.

The Hadoop version was changed from 3.3.4 to 3.3.6 because this version is used for the HBase image. This reduces the build time and the required disk space.

part of #870 

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
- [x] Changes are OpenShift compatible
- [x] All added packages (via microdnf or otherwise) have a comment on why they are added
- [x] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [x] All packages should have (if available) signatures/hashes verified
- [x] Add an entry to the CHANGELOG.md file
- [x] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

**Comment by siegfriedweber** [comment]
:hourglass_flowing_sand: https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/8/

File: spark-k8s/versions.py [hunk 1]
```diff
@@ -4,10 +4,10 @@
         "java-base": "17",
         "java-devel": "17",
         "python": "3.11",
-        "hadoop": "3.3.4",  # https://github.com/apache/spark/blob/6a5747d66e53ed0d934cdd9ca5c9bd9fde6868e6/pom.xml#L125
+        "hadoop": "3.3.6",  # Hadoop version defined in ../hbase/versions.py
```
Comment (sbernauer) [inline spark-k8s/versions.py:7]:
```suggestion
        "hadoop": "3.3.6",  # Hadoop version defined in ../hbase/versions.py to reduce build time and disk requirements
```

File: spark-k8s/versions.py [hunk 2]
```diff
@@ -22,10 +22,10 @@
...
```
Comment (sbernauer) [inline spark-k8s/versions.py:25]:
```suggestion
        "hadoop": "3.3.6",  # Hadoop version defined in ../hbase/versions.py to reduce build time and disk requirements
```

**Review by sbernauer — DISMISSED** [review]
Thanks!

**Review by sbernauer — APPROVED** [review]

**Pull request merged**